test/parallel/test-stream-writable-end-cb-error.js

```diff
@@ -46,3 +46,30 @@ const stream = require('stream');
     writable.emit('error', new Error('kaboom'));
   }));
 }
+
+{
+  const w = new stream.Writable({
+    write(chunk, encoding, callback) {
+      setImmediate(callback);
+    },
+    finish(callback) {
+      setImmediate(callback);
+    }
+  });
+  w.end('testing ended state', common.mustCall());
```

Comment:
Is this callback called with an error?

Reply:
No

Reply:
Ok... looks like it is called with an error and the way things are right now it should.

Reply:
It's like this (called with an error) also on current version 15.5.1